### PR TITLE
made width of custom url inputs flexible

### LINKED ISSUE #2311: Enlarge input field for Custom URLs
| Q | A
| --- | ---
| Bug? | yes
| New Feature? | no
| Sulu Version |  1.2.0
| Browser Version | Tested on Firefox, Chrome and Safari for Mac

#### Actual Behavior
The input field for custom URLs is very small:

custom-url

#### Expected Behavior
Fit the width of the input field to the width of the overlay.

### PR DESCRIPTION
**Pull request by marcelmoosbrugger**
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #2311
| Related issues/PRs | none
| License | MIT
| Documentation PR | none

#### What's in this PR?

In the overlay for editing custom urls, the inputs in hold prefixes and suffixes for url patterns had a constant width. The changes made, fixes this issue with the use of table visualization.

### CODE REVIEW

File: src/Sulu/Bundle/CustomUrlBundle/Resources/public/js/components/webspace/settings/custom-url/input/main.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ define(['underscore', 'jquery', 'services/husky/util'], function(_, $, util) {
 
     var defaults = {
             templates: {
-                input: '<input type="text" class="form-element<% if (!!prefix) { %> prefix" data-prefix="true"<% } else { %>" data-suffix="true"<% } %>/>'
+                input: '<div class="input-part"><input type="text" class="form-element<% if (!!prefix) { %> prefix" data-prefix="true"<% } else { %>" data-suffix="true"<% } %>/></div>'
```
Comment (danrot) [inline src/Sulu/Bundle/CustomUrlBundle/Resources/public/js/components/webspace/settings/custom-url/input/main.js]: Is that `div`-wrapper required? Or can you set the CSS properties also to `input`?

Comment (marcelmoosbrugger) [inline src/Sulu/Bundle/CustomUrlBundle/Resources/public/js/components/webspace/settings/custom-url/input/main.js]: @danrot It's probably a better idea to keep the divs for now as the Internet Explorer has some open issues on inputs and 'display: table-cell'.
https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/3419500/

File: src/Sulu/Bundle/CustomUrlBundle/Resources/public/js/components/webspace/settings/custom-url/input/main.js [hunk 2]
```diff
@@ -34,14 +34,23 @@ define(['underscore', 'jquery', 'services/husky/util'], function(_, $, util) {
...
+            while (htmlString !== tmpString) {
+                htmlString = tmpString;
+                tmpString = htmlString.replace(/^([^\/]*)\*/, '$1' + this.templates.input({prefix: true}));
+            }
+            htmlString = tmpString;
+            // replace all other "*" characters with suffix-inputs
+            htmlString = htmlString.replace(/\*/g, this.templates.input({prefix: false}));
+            // wrap all character data (data not containing ">") which is outside of divs in a domain-part div
+            htmlString = htmlString.replace(/<\/div>([^>]*)<div/g, '</div><div class="domain-part">$1</div><div');
+            // wrapp character data (data not containing ">") at the beginning in a domain-part div
```
Comment (danrot) [inline src/Sulu/Bundle/CustomUrlBundle/Resources/public/js/components/webspace/settings/custom-url/input/main.js]: wrap

**Comment by danrot** [comment]
Merged with typo fix in d722ef4

**Pull request closed without merging**